src/software/jetson_nano/services/motor.cpp

```diff
@@ -624,6 +628,40 @@ void MotorService::spiTransfer(int fd, uint8_t const* tx, uint8_t const* rx, uns
 }
 
 
+void MotorService::readThenWriteSpiTransfer(int fd, const uint8_t* read_tx,
+                                            const uint8_t* write_tx,
+                                            const uint8_t* read_rx, uint32_t spi_speed)
+{
+    int ret1, ret2;
```

Comment:
We can probably declare the variables where they're used. Though it seems like this notation was also used in `MotorService::spiTransfer`